src/utils/identity/genericSignIdentity.ts

```diff
@@ -0,0 +1,14 @@
+import { SignIdentity } from '@dfinity/agent';
+
+
+export abstract class GenericSignIdentity extends SignIdentity {
```

Comment:
I dont love this name actually we could chat with the team to improve it.

Reply:
`BaseSignIdentity` or `SignIdentityLike` would be better